app/uk/gov/hmrc/gform/validation/AddressValidation.scala

```diff
@@ -46,30 +49,50 @@ class AddressValidation[D <: DataOrigin](implicit messages: Messages, sse: Smart
         .toSeq
         .flatMap(_.toSeq)
 
+    def blankAtomicModelComponentId(atom: Atom): Option[ModelComponentId.Atomic] = {
+      val atomicFcId = fieldValue.atomicFormComponentId(atom)
+      addressValueOf(atom).filterNot(_.trim.isEmpty()).headOption.fold(Option(atomicFcId))(_ => Option.empty)
+    }
+
     def validateRequiredFieldSub(value: Atom, str: String) =
       validateRequiredField(value, str, fieldValue)(addressValueOf(value))
 
+    def validateRequiredAtom(atom: Atom, messageKey: String) =
+      blankAtomicModelComponentId(atom).fold(validationSuccess) { id =>
+        val placeholder = fieldValue.errorShortName
+          .map(_.trasform(_ + " ", identity))
```

Comment:
`trasform` this seems to be a typo, let's fix that

Reply:
It was my intention. I wanted to fit it in
```
generic.error.address.building.street.required=Enter {1}building and street
```
Here I add a space to `erroShortName` if it exists in English and I don't care about Welsh  because it doesn't have placeholders at all
```
generic.error.address.building.street.required=Nodwch adeilad a stryd
```